.github/workflows/setup-tofu.yml

```diff
@@ -0,0 +1,314 @@
+      if: runner.os != 'Windows'
+      run: |
+        cat ${HOME}/.terraformrc | grep 'credentials "tofu.example.com"'
+        cat ${HOME}/.terraformrc | grep 'token = "${{ env.TF_CLOUD_API_TOKEN }}"'
+
+  tofu-credentials-none:
+    name: 'OpenTofu No Credentials'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
+
+    - name: Setup OpenTofu
+      uses: ./
+
+    - name: Validate OpenTofu Credentials (Windows)
+      if: runner.os == 'Windows'
+      run: |
+        [[ -f ${APPDATA}/terraform.rc ]] || exit 0
+
+    - name: Validate Teraform Credentials (Linux & macOS)
+      if: runner.os != 'Windows'
+      run: |
+        [[ -f ${HOME}/.terraformrc ]] || exit 0
+
+  tofu-arguments:
+    name: 'OpenTofu Arguments'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
+
+    - name: Setup OpenTofu
+      uses: ./
+
+    - name: Check No Arguments
+      run: tofu || exit 0
+
+    - name: Check Single Argument
+      run: tofu help || exit 0
+
+    - name: Check Single Argument Hyphen
+      run: tofu -help
+
+    - name: Check Single Argument Double Hyphen
+      run: tofu --help
+
+    - name: Check Single Argument Subcommand
+      run: tofu fmt -check
+
+    - name: Check Multiple Arguments Subcommand
+      run: tofu fmt -check -list=true -no-color
+
+  tofu-arguments-no-wrapper:
+    name: 'OpenTofu Arguments No Wrapper'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
+
+    - name: Setup OpenTofu
+      uses: ./
+      with:
+        tofu_wrapper: false
+
+    - name: Check No Arguments
+      run: tofu || exit 0
+
+    - name: Check Single Argument
+      run: tofu help || exit 0
+
+    - name: Check Single Argument Hyphen
+      run: tofu -help
+
+    - name: Check Single Argument Double Hyphen
+      run: tofu --help
+
+    - name: Check Single Argument Subcommand
+      run: tofu fmt -check
+
+    - name: Check Multiple Arguments Subcommand
+      run: tofu fmt -check -list=true -no-color
+
+  tofu-run-local:
+    name: 'OpenTofu Run Local'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    defaults:
+      run:
+        working-directory: ./.github/workflows/data/local
+    steps:
+    - name: Checkout
+      uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
+
+    - name: Setup OpenTofu
+      uses: ./
+
+    - name: OpenTofu Init
+      shell: bash
```

Comment:
Since we have specified `defaults: run: shell: bash` at the top-level, do we need to re-specify `shell: bash` again in subsequent jobs' steps?

Reply:
It's a copy-paste relict from setup-terraform mate 🤷‍♂️ Thank you for noticing!

Reply:
Implemented